pkg/controller/ec2keypair/ec2keypair_controller.go

```diff
@@ -255,6 +256,13 @@ func (r *ReconcileEC2KeyPair) Reconcile(request reconcile.Request) (reconcile.Re
 					instance.ObjectMeta.Finalizers[i+1:]...)
 			}
 		}
+
+		//check for finalizers not placed by the controller
+		if len(instance.ObjectMeta.Finalizers) != 0 {
+			r.events.Eventf(instance, `Warning`, `DeleteFailure`, "Unable to delete the EC2SecurityGroup with remaining finalizers")
```

Comment:
What does this have to do with "EC2SecurityGroups?"

Reply:
This was due to some less than stellar copy & pasting. Fixed.